### DISCUSSION TITLE
Getting the residuals of the iterative solver

### DISCUSSION THREAD

**Discussion by KapilKhanal** [capytaine/capytaine#161] [Q&A] [answered]
Hi, I am trying to get the residuals from the iterative solver used in Capytaine. How to go about getting those for each step ?
Thanks

**Comment by mancellin** [comment, marked as answer]
You could define a custom linear solver as follow

```python
def my_linear_solver(A, b):
    # Do here whatever you want to solve Ax = b
    # For instance, the current default is basically:
    from scipy.sparse.linalg import gmres
    x, _ = gmres(A, b)
    return x

solver = cpt.BEMSolver(engine=cpt.BasicMatrixEngine(linear_solver=my_linear_solver))
... = solver.solve(...)
```

Then it is up to you to figure out how to make Scipy's `gmres` in `my_linear_solver` do what you want, or replace it by something else.
